charts/jenkins/templates/cache-cronjobs.yaml

```diff
@@ -1,38 +1,32 @@
-{{- /* create map of agents by clearJob.componentName */}}
-{{- if .Values.agent.cache.clearJob.create }}
-  {{- $_ := set .Values "agentMap" (dict .Values.agent.cache.clearJob.componentName .Values.agent) }}
-{{- else }}
-  {{- $_ := set .Values "agentMap" (dict) }}
-{{- end }}
+{{- /* create agent map */}}
+{{- /* merge .Values.agent into additional agent(s) to propogate values except cache.clearJob.create */}}
+{{- $_ := set .Values "agentMap" (dict "agent" .Values.agent) }}
+{{- $agentCopy := deepCopy .Values.agentMap.agent }}
+{{- $_ := set $agentCopy.cache.clearJob "create" false }}
 {{- range $name, $additionalAgent := .Values.additionalAgents }}
-    {{- /* merge .Values.agent into additional agent to ensure it at least has the default values */}}
-    {{- $mergedAgent := (merge $additionalAgent $.Values.agent) }}
-    {{- if $mergedAgent.cache.clearJob.create }}
-      {{- if get $.Values.agentMap $mergedAgent.cache.clearJob.componentName }}
-        {{- fail "Cannot create more than one clearJob with a given componentName" }}

```

Comment:
I think it'll be ok for users to receive the  kubernetes "duplicate resource" error as I think the cache values are intuitive enough where the user will realize they need to set/modify `componentName`. Mostly just trying to avoid the extra templating.